bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/EGateHandler.java

```diff
@@ -0,0 +1,303 @@
+     * @throws InterruptedException
+     *
+     */
+
+    public void tiltUp(String id) throws InterruptedException {
+        for (int i = 0; i < 3; i++) {
+            pulseUp(id);
+            Thread.sleep(150); // sleep to not confuse the blinds
+
+        }
+    }
+
+    public void tiltDown(String id) throws InterruptedException {
+        for (int i = 0; i < 3; i++) {
+            pulseDown(id);
+            Thread.sleep(150);// sleep to not confuse the blinds
+        }
+    }
+
+    public void pulseUp(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=1;Priority=1;CheckNr=3415347;" + CR);
+    }
+
+    public void pulseDown(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=2;Priority=1;CheckNr=2764516;" + CR);
+    }
+
+    public void continuousUp(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=3;Priority=1;CheckNr=2867016;" + CR, 20000);
+    }
+
+    public void continuousDown(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=4;Priority=1;CheckNr=973898;" + CR, 20000);
+    }
+
+    public void stop(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=5;Priority=1;CheckNr=5408219;" + CR);
+    }
+
+    public void registerBlind(String id, ThingUID uid) {
+        logger.debug("Registring Blind id {} with thingUID {}", id, uid);
+        registeredBlinds.put(id, uid);
+    }
+
+    /**
+     * Send a command to the eGate Server.
+     */
+
+    private void sendCommand(String command) {
+        sendCommand(command, SOCKET_TIMEOUT_SEC);
+    }
+
+    private synchronized void sendCommand(String command, int timeout) {
+        logger.debug("EGate got command: {}", command);
+        Socket localEGateSocket = egateSocket;
+        BufferedWriter localWriter = writer;
+        if (localEGateSocket == null || localWriter == null) {
+            return;
+        }
+        if (!isConnected()) {
+            logger.debug("no connection to Dominoswiss eGate server when trying to send command, returning...");
+            return;
+        }
+
+        // Send plain string to eGate Server,
+        try {
+            localEGateSocket.setSoTimeout(SOCKET_TIMEOUT_SEC);
+            localWriter.write(command);
+            localWriter.flush();
+        } catch (IOException e) {
+            logger.debug("Error while sending command {} to Dominoswiss eGate Server {} ", command, e.toString());
+        }
+    }
+
+    private void pollingConfig() {
+        if (!isConnected()) {
+            Socket localSocket = egateSocket;
+
+            synchronized (lock) {
+                try {
+                    localSocket = new Socket();
+                    localSocket.connect(new InetSocketAddress(host, port));
+                    localSocket.setSoTimeout(SOCKET_TIMEOUT_SEC);
+                    BufferedWriter localWriter = writer;
+                    localWriter = new BufferedWriter(new OutputStreamWriter(localSocket.getOutputStream()));
```

Comment:
You need to close the BufferedWriter. You could use the auto-close feature:
```suggestion
                    try(localWriter = new BufferedWriter(new OutputStreamWriter(localSocket.getOutputStream()))) {
```

If I see correctly, you create new BufferedWriter/OutputStreamWriter each poll. Why not create one when connecting and close it in dispose() or disconnect? Then, you can also re-use it in sendCommand().

Reply:
I close the BufferedWriter. The redesign will be made with a later release, thanks for that hint.

Reply:
The BufferedWriter is assigned to the local variable `localWriter`, but this variable is out of scope if the method `pollingConfig` is left. So, the BufferedWrite is never closed and you will end up with a resource leak.

As said, the assignment `BufferedWriter localWriter = writer;` has no effect.